### PR TITLE
align environment variable name with docusaurus instructions

### PR DESCRIPTION
**Pull request by SethTisue**

### CODE REVIEW

File: .github/workflows/publish-website.yml [hunk 1]
```diff
@@ -15,4 +15,4 @@ jobs:
       - name: Publish ${{ github.ref }}
         run: sbt docs/docusaurusPublishGhpages
         env:
-          GITHUB_DEPLOY_KEY: ${{ secrets.GITHUB_DEPLOY_KEY }}
+          GIT_DEPLOY_KEY: ${{ secrets.GIT_DEPLOY_KEY }}
```
Comment (danicheg) [inline .github/workflows/publish-website.yml:18]: I think it's no matter how we define the env var in secrets, but I thought it should be `GITHUB_DEPLOY_KEY` in CI as done in Scalafix, for example https://github.com/scalacenter/scalafix/blob/main/.github/workflows/deploy-website.yml#L17

Comment (danicheg) [reply .github/workflows/publish-website.yml:18]: Yeah, I'm pretty confident it should be `GITHUB_DEPLOY_KEY` in CI https://github.com/search?q=org%3Ascalacenter%20GITHUB_DEPLOY_KEY&type=code

Comment (danicheg) [reply .github/workflows/publish-website.yml:18]: Perhaps the mdoc website is a little bit outdated 🤷🏻

**Comment by danicheg** [comment]
yay, it's alive https://scalacenter.github.io/scalac-profiling/

File: .github/workflows/publish-website.yml (hunk 1 above)
Comment (SethTisue) [reply .github/workflows/publish-website.yml]: can't think why it would matter what it's called as long as we're consistent 🤷

**Comment by SethTisue** [comment]
yay

**Pull request merged**